### PR TITLE
Module Added: sccm_status

### PR DESCRIPTION
**Pull request by computeronix**
Added module to report on sccmstatus

The credit for the original module should go to https://github.com/hunty1/MunkiReport_modules

I mainly pulled it and modified bits and pieces to get it (partially) working

The module works and pulls and stores the sccmstatus information properly

However: Please note I do need help on the listing views
Bugs:
In the list it shows as blank instead of SCCM Status as an option to select from

When you select the blank it loads the listing with errors (I am assuming it is using the old way to generate the listing (2015) and the new way has not yet been modified).  I attempted modification but could not get it to display properly

Anyone with experience with the listing views if you could review and modify what is needed to get it to work

### CODE REVIEW

File: app/modules/sccmstatus/sccm_status_model.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                $this->rs['agent_status'] = '';
+                $this->rs['mgmt_point'] = '';
+                $this->rs['enrollment_name'] = '';
+                $this->rs['enrollment_server'] = '';     
+                $this->rs['last_checkin'] = '';
+                $this->rs['cert_exp'] = '';       
+
+                // Schema version, increment when creating a db migration
+                $this->schema_version = 0;
+
```
Comment (bochoven) [inline app/modules/sccmstatus/sccm_status_model.php]: You need to add indexes for columns you are going to need to search or sort on. Which usually is every column apart from id and serial_number.

File: app/views/listing/sccm_status.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+?>
+
+<div class="container">
+
+  <div class="row">
+
+  	<div class="col-lg-12">
+		<script type="text/javascript">
+
+		$(document).ready(function() {
```
Comment (bochoven) [inline app/views/listing/sccm_status.php]: Please start with a current listing view, this code is old.

File: app/views/listing/sccm_status.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+			    
+			} );
+		</script>
+
+		  <h3>SCCM Agent Status report <span id="total-count" class='label label-primary'>…</span></h3>
+
+		  <table class="table table-striped table-condensed table-bordered">
+		    <thead>
+		      <tr>
+		      	<th data-colname='machine#computer_name'>Name</th>
```
Comment (bochoven) [inline app/views/listing/sccm_status.php]: Please localise the column headers (see other listings)

**Comment by computeronix** [comment]
Thanks for the comments!! Will review and adjust as recommended

File: app/modules/sccmstatus/sccm_status_model.php (hunk 1 above)
Comment (computeronix) [inline app/modules/sccmstatus/sccm_status_model.php]: Attempted fix implemented

**Comment by computeronix** [comment]
added indexing to: filevault_escrow_model.php

meant to put that in a different pull request

**Comment by computeronix** [comment]
Ok moved things around and made adjustments to the locales and to the files

Try out the module now

Bug:
When you click on a machine within the module listing that has information it takes you to the tab page but does not show the information :(    if you have ideas with that let me know :)

File: app/modules/filevault_escrow/filevault_escrow_model.php [hunk 4]
```diff
@@ -19,7 +19,12 @@ public function __construct($serial = '')
         $this->rs['HddSerial'] = '';
 
         // Schema version, increment when creating a db migration
-        $this->schema_version = 0;
+        $this->schema_version = 1;
```
Comment (bochoven) [inline app/modules/filevault_escrow/filevault_escrow_model.php]: Don't make changes to other modules, that gets confusing very quickly

File: app/modules/sccm_status/sccm_status_controller.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ * SCCM Agent status module class
+ *
+ * @package munkireport
+ * @author
+ **/
+class sccm_status_controller extends Module_controller
+{
+        
+        /*** Protect methods with auth! ****/
+        function __construct()
```
Comment (bochoven) [inline app/modules/sccm_status/sccm_status_controller.php]: Please add the `public` keyword for all methods in a class

File: app/modules/sccm_status/sccm_status_controller.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+                // Store module path
+                $this->module_path = dirname(__FILE__);
+        }
+
+        /**
+         * Default method
+         *
+         * @author Calum Hunter
+         **/
+        function index()
```
Comment (bochoven) [inline app/modules/sccm_status/sccm_status_controller.php]: `public`

File: app/modules/sccm_status/scripts/sccm_status_info.sh [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+# Create cache dir if it does not exit
+DIR=$(dirname $0)
+mkdir -p "$DIR/cache"
+
+# Location of our output
+sccm_file="/usr/local/munki/preflight.d/cache/sccm_status.txt"
+
+# Location of the SCCM Plist
+ccmpref="/var/root/Library/Preferences/com.microsoft.ccmclient.plist"
```
Comment (bochoven) [inline app/modules/sccm_status/scripts/sccm_status_info.sh]: For maximum compliance, leave off the .plist extension

File: app/views/listing/sccm_status.php [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+		  <table class="table table-striped table-condensed table-bordered">
+		    <thead>
+		      <tr>
+		      	<th data-i18n="listing.computername" data-colname='machine.computer_name'></th>
+		        <th data-i18n="serial" data-colname='machine.serial_number'></th>
+		        <th data-i18n="sccm_status.agent_status" data-colname='sccm_status.agent_status'></th> 
+		        <th data-i18n="sccm_status_.mgmt_point" data-colname='sccm_status.mgmt_point'></th>
+		        <th data-i18n="sccm_status.enrollment_name" data-colname='sccm_status.enrollment_name'></th>
+		        <th data-i18n="sccm_status.enrollment_server" data-colname='sccm_status.enrollment_server'></th>
+				<th data-i18n="sccm_status.last_checkin" data-colname='sccm_status.last_checkin'></th>
```
Comment (bochoven) [inline app/views/listing/sccm_status.php]: please fix indentation

File: app/views/client/sccm_status_tab.php [hunk 9]
```diff
@@ -0,0 +1,43 @@
+<?php //Initialize models needed for the table
+$sccm_status = "";
+?>
+	<h2>SCCM_Agent_Status</h2>
+
+		<table class="table table-striped">
+			<tbody>
+			<thead>
+				<tr>
+					<th>SCCM Agent Information:</th>
```
Comment (bochoven) [inline app/views/client/sccm_status_tab.php]: Please use i18n to localise these headings

File: app/views/client/sccm_status_tab.php [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+					<th>Value</th>
+				</tr>
+			</thead>
+				<tr>
+					<td>Enrollment Status</td>
```
Comment (bochoven) [inline app/views/client/sccm_status_tab.php]: Please use i18n to localise these headings

File: app/modules/filevault_escrow/filevault_escrow_model.php (hunk 4 above)
Comment (bochoven) [inline app/modules/filevault_escrow/filevault_escrow_model.php]: If you want to improve the file vault_escrow module, please create another pull request

File: app/modules/sccm_status/scripts/sccm_status_info.sh (hunk 7 above)
Comment (computeronix) [inline app/modules/sccm_status/scripts/sccm_status_info.sh]: done

File: app/views/client/sccm_status_tab.php (hunk 9 above)
Comment (computeronix) [inline app/views/client/sccm_status_tab.php]: done

File: app/views/client/sccm_status_tab.php (hunk 10 above)
Comment (computeronix) [inline app/views/client/sccm_status_tab.php]: done

**Comment by computeronix** [comment]
all changes have been made

one question -- any ideas why the sccm information is not going into the table?  The SCCM text file gets created and the sccm model information looks correct but the entry does not go in?   Dont see anything in the logs as failures or warnings

**Comment by computeronix** [comment]
ok saw the first time it ran this

Oct 24 2016 14:56:34 -0400     Munkireport: Requesting sccm_status
Oct 24 2016 14:56:36 -0400     Munkireport: Need to update sccm_status
Server info: starting: sccm_status


this has never flagged again even though the table is deleted; is there a value that needs deleted in another table for this to recognize to update again?

**Comment by bochoven** [comment]
@computeronix Munkireport stores a md5 hash of the data: https://github.com/munkireport/munkireport-php/wiki/How-to-create-a-module#caching

**Comment by computeronix** [comment]
thanks for your fix and caching information!

made another fix and it works now!

**Comment by computeronix** [comment]
One last item: the locales was working but when I cleared my install and tested the newly added code -- locales are not working and just show blanks for the sccm stuff

any ideas?  I didnt see a character missplacement or anything

**Comment by bochoven** [comment]
Concerning my last commit: I want to remove all php code from the views and only use ajax calls to populate data.

**Comment by bochoven** [comment]
Not sure what I did, but your code got merged! Thanks!

**Pull request merged**